bolt-sidecar/src/state/execution.rs

```diff
@@ -311,8 +318,11 @@ impl<C: StateFetcher> ExecutionState<C> {
             return Err(ValidationError::BaseFeeTooLow(max_basefee));
         }
 
-        // Ensure max_priority_fee_per_gas is greater than or equal to min_priority_fee
-        if !req.validate_min_priority_fee(max_basefee, self.limits.min_priority_fee) {
+        // Create pricing calculator
+        let pricing = pricing::PreconfPricing::new(self.validation_params.block_gas_limit);
```

Comment:
I think the pricing calculator could be part of the execution state itself instead of being recreated every time

Reply:
agreed!